include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -39,7 +39,7 @@ struct NodeComputeInfo {
   DestroyFunctionStateFunc release_state_func;
 };
 
-class IExecutionProvider {
+class IExecutionProvider : public IAllocatorManager {
```

Comment:
I am not sure if there's any is-a relationship between a provider and the allocator mgr.

Reply:
Maybe I can have a better name for it. The function in IAllocatorManager  were extracted from IExecutionProvider. The major purpose is to allow create a kernel without creating an EP or KernelRegistry.

Reply:
Reverted